app/src/org/commcare/views/widgets/MediaWidget.java

```diff
@@ -123,24 +123,37 @@ public IAnswerData getAnswer() {
     }
 
     /**
-     * @return resolved filepath or null if the target is too big to upload
+     * @return whether the media file passes the size check
      */
-    private String getBinaryPathWithSizeCheck(Object binaryURI) throws IOException {
-        String binaryPath = createFilePath(binaryURI);
+    private boolean ifMediaSizeChecks(String binaryPath) {
         File source = new File(binaryPath);
         boolean isTooLargeToUpload = checkFileSize(source);
-
-        if (mBinaryName != null) {
-            deleteMedia();
-        }
-
         if (isTooLargeToUpload) {
             oversizedMediaSize = (int)source.length() / (1024 * 1024);
-            return null;
+            return false;
         } else {
             oversizedMediaSize = -1;
-            return binaryPath;
+            return true;
+        }
+    }
+
+    /**
+     * @return whether the media file has a valid extension
+     */
+    private boolean ifMediaExtensionChecks(String binaryPath) {
+        String extension = FileUtil.getExtension(recordedFileName);
+        if (!FormUploadUtil.isSupportedMultimediaFile(binaryPath)) {
+            Toast.makeText(getContext(),
+                    Localization.get("form.attachment.invalid"),
+                    Toast.LENGTH_LONG).show();
+            Log.e(TAG, String.format(
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/20388d5a2c6618e54b8fddc7da3bc87f0b52cff3/app/src/org/commcare/views/widgets/MediaWidget.java#L149"></span>Using 'Log' instead of 'Timber'</td>
    </tr>
  </tbody>
</table>